app/Traits/HasStructuredData.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+
+namespace App\Traits;
+
+
+use Spatie\SchemaOrg\Type;
+
+/**
+ * Trait HasStructuredData to implement a builder for all Schema.org types
+ * and their properties using the spatie/schema-org package.
+ *
+ * @see https://schema.org
+ * @see https://github.com/spatie/schema-org
+ *
+ * @package App\Traits
+ */
+trait HasStructuredData
+{
+    /**
+     * Factory method for generating the script for a Schema.org type.
+     *
+     * @return Type
+     */
+    abstract protected function generateStructuredDataScript() : Type;
```

Comment:
This is the abstract method each model has to implement (using spatie/schema-org library).